l10n_be_cooperator_national_number/views/subscription_request_view.xml

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+
+
+    <record id="subscription_request_view_form" model="ir.ui.view">
+        <field name="name">subscription.request.view.form</field>
+        <field name="model">subscription.request</field>
+        <field name="inherit_id" ref="cooperator.subscription_request_view_form" />
+        <field name="arch" type="xml">
+            <field name="country_id" position="after">
+                <field
+                    name="national_number"
+                    attrs="{'invisible': ['|', ('display_national_number', '=', False), ('is_company', '=', True)],
+                'required': [('display_national_number', '=', True),
+                ('is_company', '=', False)]}"
```

Comment:
Maybe this should be `[('company_id.require_national_number', '=', True), ('is_company', '=', False)]` (or wrap that domain into its own computed field `require_national_number`)

Reply:
Yes creating a computed field is the way to go I think.